tests/php/ORM/DataQueryFixedFieldsTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\ORM\Tests;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataQuery;
+
+class DataQueryFixedFieldsTest extends SapphireTest
+{
+    protected static $fixture_file = 'DataQueryFixedFieldsTest.yml';
+
+    protected static $extra_dataobjects = [
+        DataQueryTest\ObjectA::class,
+    ];
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        DataObject::config()->merge('fixed_fields', ['ExtraFixedField' => 'Varchar']);
+        static::tempDB()->resetDBSchema(static::$extra_dataobjects);
+    }
+
+    public function testDataQueryHasFixedFields()
+    {
+        $dataQuery = new DataQuery(DataQueryTest\ObjectA::class);
+        $dataQuery->setQueriedColumns(['Name']);
+        $this->assertSame(['This is the field'], $dataQuery->execute()->column('ExtraFixedField'));
```

Comment:
Looks good to me. 
Just small question, could we or do we need to check, just in case, that "ClassName" etc. is still available in `fixed_fields `?

Reply:
I don't think so - if those aren't available anymore that would be a failure in config which has its own unit tests.